File: 1-Patron/python/GestorDeUltimate.py
import Ultimate
class GestorDeUltimate():
    def __init__(self, ultimate):
        self.__ultimate = ultimate

    def subirDeNivel(self, incremento):
        text = 'No se pudo subir de nivel la ultimate'
        exito = self.__ultimate.incNivel(incremento)
        if (exito):
            text = 'Nuevo nivel alcanzado'
        return text
        
    def generarUltimate():
        return Ultimate("Basic Ulti", 10, 1)

File: 2-Patron/Java/GestordeUltimate.java
package abstractmasterwhole;

public class GestordeUltimate {
    private Ultimate ultimate;

    public GestordeUltimate(Ultimate ultimate) {
        this.ultimate = ultimate;
    }

    public int subirDeNivel(int incremento){
        return this.ultimate.incNivel(incremento);
    }
    
    public Ultimate generarUltimate(){
        return new Ultimate("basic",10,1);
    }
}
